### PR TITLE
add IsMockResourcePool()

### PR DESCRIPTION
**Pull request by okigan**

### CODE REVIEW

File: resource/resources.go [hunk 1]
```diff
@@ -11,4 +11,9 @@ const (
 	ResourceNameDisk          = "ephemeral-storage"
 	ResourceNameDiskLegacy    = "storage"
 	ResourceNamePods          = "pods"
+	ResourceNameMockNodesPool = "mock-nodes"
```
Comment (solarkennedy) [inline resource/resources.go:14]: This shouldn't go in resources.go, wrong kind of `resource`.

make `resourcepools/resourcepools.go`

Comment (solarkennedy) [inline resource/resources.go:14]: `ResourcePoolMockPods`

Would pair well with

`ResourcePoolElastic`
`ResourcePoolReserved`
`ResourcePoolG4dnMetal`

which are currently strings hanging out elsewhere

Comment (okigan) [reply resource/resources.go:14]: making resourcepool (keeping singular to match pod and node)

File: resourcepool/resourcepool.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package resourcepool
+
+const (
+	ResourcePoolMockPods = "mock-nodes"
```
Comment (solarkennedy) [inline resourcepool/resourcepool.go:4]: MockNodes

File: resourcepool/resourcepool.go [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package resourcepool
+
+const (
```
Comment (solarkennedy) [inline resourcepool/resourcepool.go:3]: Can you also add in more constants so we can get rid of other constants.go spread across:

```
	// Resource Pools
	Elastic    = "elastic"
	ElasticGPU = "elasticGpu"
	Reserved   = "reserved"
```

File: resourcepool/resourcepool.go [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+	ResourcePoolElastic    = "elastic"
+	ResourcePoolElasticGPU = "elasticGpu"
+	ResourcePoolReserved   = "reserved"
+	ResourcePoolMocNodes   = "mock-nodes"
```
Comment (solarkennedy) [inline resourcepool/resourcepool.go:7]: Mock

File: resourcepool/resourcepool.go [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package resourcepool
+
+const (
+	ResourcePoolElastic    = "elastic"
+	ResourcePoolElasticGPU = "elasticGpu"
```
Comment (solarkennedy) [inline resourcepool/resourcepool.go:5]: No `ResourcePoolElasticGPU`, I don't know why we have that

**Review by solarkennedy — APPROVED** [review]

**Pull request merged**